### PR TITLE
chore: remove jwt incorrect key warning

### PR DESCRIPTION
**Pull request by vishwarajanand**
fixes https://github.com/firebase/php-jwt/issues/559

### CODE REVIEW

File: src/JWT.php [hunk 1]
```diff
@@ -251,6 +251,9 @@ public static function sign(
                 return \hash_hmac($algorithm, $msg, $key, true);
             case 'openssl':
                 $signature = '';
+                if (!openssl_pkey_get_private($key)) {
```
Comment (vishwarajanand) [inline src/JWT.php:254]: The method signature has `resource` type hint, which mismatches with the `openssl_pkey_get_private`. I see it was added as part of https://github.com/firebase/php-jwt/pull/405

Comment (ajupazhamayil) [reply src/JWT.php:254]: Can we add resource to the called function to resolve this issue?

Comment (vishwarajanand) [reply src/JWT.php:254]: Not east to do now. I think the fix is likely going to be a backwards incompatible one. 
Else, we can just ignore the warning here.

Comment (vishwarajanand) [reply src/JWT.php]: I made a hacky fix by skipping private key check for resource type, but I am not sure whether this is the most appropriate fix.

File: src/JWT.php [hunk 2]
```diff
@@ -251,6 +251,9 @@ public static function sign(
                 return \hash_hmac($algorithm, $msg, $key, true);
             case 'openssl':
                 $signature = '';
+                if (!\is_resource($key) && !openssl_pkey_get_private($key)) {
```
Comment (noahdietz) [inline src/JWT.php:254]: So if I understand correctly, we want to validate that if `$key` is not a PHP Resource, we can still call `openssl_pkey_get_private ` on it (b.c I assume the following call to `openssl_sign` makes this same call under the hood)?

Comment (vishwarajanand) [reply src/JWT.php:254]: Yes, that's correct.

Comment (noahdietz) [reply src/JWT.php:254]: Seems fine then! I wonder if there is a more useful error message we can produce from said validation, but if not, this is fine.

**Review by noahdietz — APPROVED** [review]

**Review by ajupazhamayil — APPROVED** [review]

**Pull request merged**